esmvaltool/cmorizers/data/formatters/datasets/esacci_landcover.py

```diff
@@ -0,0 +1,378 @@
+
+
+def regrid_fix(cube, vals, glob_attrs, var_name, var_info):
+    """Regrid cube and fixes.
+
+    Regrids the cube, adds typebare coordinate and fix coordinates.
+
+    Parameters
+    ----------
+    cube: iris.cube.Cube
+          Data cube to be regridded.
+
+    vals: dict
+          Variable long_name.
+
+    glob_attrs: dict
+          Dictionary holding cube metadata attributes.
+
+    var_name: str
+          Variable name.
+
+    var_info: dict
+          Dictionary holding cube metadata attributes.
+
+    Returns
+    -------
+    cube: iris.cube.Cube
+        data cube regreidded and with fixed coordinates.
+    """
+    logger.info("Regridding cube for %s", var_name)
+    regridded_cube = regrid_iris(cube)
+    fix_var_metadata(regridded_cube, var_info)
+    if vals['long_name'] == 'BARE':
+        add_typebare(regridded_cube)
+    regridded_cube = fix_coords(regridded_cube)
+    set_global_atts(regridded_cube, glob_attrs)
+
+    return regridded_cube
+
+
+def cmorization(in_dir, out_dir, cfg, cfg_user, start_date, end_date):
+    """Cmorize data."""
+    glob_attrs = cfg['attributes']
+    if not start_date:
+        start_date = datetime(1992, 1, 1)
+    if not end_date:
+        end_date = datetime(1992, 12, 31)
+
+    shrub_vars = {'shrubs-bd', 'shrubs-be', 'shrubs-nd', 'shrubs-ne'}
+    shrub_cubes = []
+    tree_vars = {'trees-bd', 'trees-be', 'trees-nd', 'trees-ne'}
+    tree_cubes = []
+
+    for var_name, vals in cfg['variables'].items():
+        if not isinstance(vals, dict):
+            raise ValueError(f"Invalid format for variable {var_name}: "
+                             f"{type(vals)}")
+        var_info = cfg['cmor_table'].get_variable(vals['mip'], var_name)
+        glob_attrs['mip'] = vals['mip']
+        raw_info = {'name': vals['long_name']}
+        inpfile_pattern = os.path.join(in_dir, vals['filename'])
+        logger.info("CMORizing var %s from file type %s", var_name,
+                    inpfile_pattern)
+
+        for year in range(start_date.year, end_date.year + 1):
+            year_inpfile_pattern = inpfile_pattern.format(year=year)
+            inpfiles = sorted(glob.glob(year_inpfile_pattern))
+            for inpfile in inpfiles:
+                raw_info['file'] = inpfile
+                cube_list = extract_variable(raw_info)
+                for cube in cube_list:
+                    if var_name in shrub_vars:
+                        shrub_cubes.append(cube)
+                    elif var_name in tree_vars:
+                        tree_cubes.append(cube)
+                    else:
+                        regridded_cube = regrid_fix(cube, vals, glob_attrs,
+                                                    var_name, var_info)
+                        save_variable(regridded_cube, var_name, out_dir,
+                                      glob_attrs,
+                                      unlimited_dimensions=['time'])
```

Comment:
Since the filename is the same for all variables, it's much more efficient to first iterate over the years and the the variables. This way you only have to load the files once, e.g.

```suggestion
    for year in range(start_date.year, end_date.year + 1):
        year_inpfile_pattern = inpfile_pattern.format(year=year)
            inpfiles = sorted(glob.glob(year_inpfile_pattern))
            for inpfile in inpfiles:
                cubes = iris.load(inpfile)
                for var_name, vals in cfg['variables'].items():
                    var_info = cfg['cmor_table'].get_variable(vals['mip'], var_name)
                    glob_attrs['mip'] = vals['mip']
                    if var_name == 'shrubFrac':
                        cube = cubes.extract_cube('SHRUBS-BD') + ...
                    elif var_name == 'treeFrac':
                        cube = cubes.extract_cube('TREES-BD') + ...
                    else:
                        cube = cubes.extract_cube(vals['long_name'])
                    if var_name == 'baresoil_frac':
                        add_typebare(cube)
                    regridded_cube = regrid_fix(cube, vals, glob_attrs, var_name, var_info)
                        save_variable(
                            regridded_cube, var_name, out_dir, glob_attrs, unlimited_dimensions=['time']
                        )
                    
                        
                    
                    
                        
```